### PR TITLE
feat: Add changeRequests Alerts

### PR DESCRIPTION
**Pull request by fpigeonjr**
## What changed

- Adds success Alert to submitters (no changeRequests) 😞 
- Adds Alert for all users with ChangeRequests 🤩 

## Issue

#2193 #2323 #2324 

## How to test

1. Make changes to Agreement 2 budget-lines
2. Change amount, need-by-date, and or CAN
3. Save changes
4. See confirmation modal.
5. See Success Alert for submitter and after timeout see Alert for all users

## Screenshots

image

### CODE REVIEW

File: frontend/src/hooks/useChangeRequests.hooks.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { useGetAgreementByIdQuery, useGetCansQuery } from "../api/opsAPI";
```
Comment (fpigeonjr) [inline frontend/src/hooks/useChangeRequests.hooks.js:1]: this is the crux of the PR. This hook sends off an array of properly formatted changeRequests to the AgreementChangeRequestAlert

File: frontend/src/components/Agreements/AgreementChangesAlert/AgreementsChangesAlert.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Alert for when there are agreement changes in review.
+ * @component
+ * @param {Object} props - The component props.
+ * @param {string[]} props.changeRequests - The change requests.
+ * @param {boolean} props.isAlertVisible - Whether the alert is visible.
+ * @param {Function} props.setIsAlertVisible - The function to set the alert visibility.
+ * @returns {JSX.Element} - The rendered component.
+ */
+function AgreementChangesAlert({ changeRequests, isAlertVisible, setIsAlertVisible }) {
```
Comment (fpigeonjr) [inline frontend/src/components/Agreements/AgreementChangesAlert/AgreementsChangesAlert.jsx:13]: pass in changeRequests

File: frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js [hunk 3]
```diff
@@ -131,16 +132,21 @@ const useCreateBLIsAndSCs = (
...
                     });
-                    resetForm();
-                    setIsEditMode(false);
-                    setAlert({
-                        type: "success",
-                        heading: "Agreement Edits Sent to Approval",
-                        message:
-                            "Your edits have been successfully sent to your Division Director to review. After edits are approved, they will update on the Agreement",
-                        redirectUrl: `/agreements/${selectedAgreement?.id}`
+                    Promise.all(promises).then(() => {
```
Comment (johndeange) [inline frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js:140]: You may want to also consider if one of the promises are rejected here

Comment (fpigeonjr) [reply frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js]: https://github.com/HHS/OPRE-OPS/assets/4629398/8aea82bc-b137-41f2-84bb-752b8cc109dc

Comment (fpigeonjr) [reply frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js]: solved with the Promise.allSettled

```js
 Promise.allSettled(promises).then((results) => {
                        let rejected = results.filter((result) => result.status === "rejected");
                        if (rejected.length > 0) {
                            console.error(rejected[0].reason);
                            setAlert({
                                type: "error",
                                heading: "Error Sending Agreement Edits",
                                message: "There was an error sending your edits for approval. Please try again.",
                                redirectUrl: "/error"
                            });
                        } else {
                            resetForm();
                            setIsEditMode(false);
                            setAlert({
                                type: "success",
                                heading: "Agreement Edits Sent to Approval",
                                message:
                                    "Your edits have been successfully sent to your Division Director to review. After edits are approved, they will update on the Agreement",
                                redirectUrl: `/agreements/${selectedAgreement?.id}`
                            });
                        }
```

File: frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js [hunk 4]
```diff
@@ -409,6 +423,8 @@ const useCreateBLIsAndSCs = (
                     }
                     navigate(`/agreements/${selectedAgreement?.id}/budget-lines`);
                 } else {
+                    // TODO: Add logic to delete the agreement in the workflow
```
Comment (rajohnson90) [inline frontend/src/components/BudgetLineItems/CreateBLIsAndSCs/CreateBLIsAndSCs.hooks.js:426]: Is this TODO required anymore? It looks like you did add the logic to delete the agreement in the lines below.

**Review by rajohnson90 — APPROVED** [review]
The PR is good. Outside my one question, there's no issues. Put the approval down because I don't think the one TODO statement is enough to prevent approval.

**Pull request merged**